src/Gjallarhorn/State.fs

```diff
@@ -82,3 +82,39 @@ type State<'TModel,'TMsg> (initialState : 'TModel, update : 'TMsg -> 'TModel ->
         member this.Dispose() = 
             (signal :?> System.IDisposable).Dispose()            
             GC.SuppressFinalize this
+
+/// A wrapper for a mutable value with change notification, based on/derived from clojure's Atoms
+type AtomicMutable<'a>(value : 'a) =
+    let v = ref value
+
+    let rec setValue newValue =
+        let result = Interlocked.CompareExchange<'a>(v, newValue, !v)
+        if obj.ReferenceEquals result !v then ()
+        else Thread.SpinWait 20; setValue newValue
+
+    // Stores dependencies remotely to not use any space in the object (no memory overhead requirements)
+    member private this.Dependencies with get() = Dependencies.createRemote this
```

Comment:
You'd just pass in an empty list to start.  They get added when things subscribe.  SignalBase needs that because it gets constructed from other signals - so it starts with dependencies.